src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/prisonperson/PrisonPersonSyncIntTest.kt

```diff
@@ -1,36 +1,439 @@
+                  aPhysicalAttributesResponse(
+                    heightCentimetres = 170,
+                    weightKilograms = 70,
+                  ),
+                ),
+              ),
+            ),
+          ),
+        )
+        dpsPrisonPersonServer.stubSyncPrisonPerson(aPhysicalAttributesHistoryDto())
+
+        sendPhysicalAttributesChangedEvent(bookingId = 12344)
+
+        verifyResults(
+          bookingId = 12344,
+          dpsUpdates = mapOf(
+            "height" to equalTo("180"),
+            "weight" to equalTo("80"),
+            "appliesFrom" to equalTo("${now.minusDays(3).toZoned()}"),
+            "appliesTo" to equalTo("${now.minusDays(2).toZoned()}"),
+          ),
+        )
+      }
+    }
+
+    @Nested
+    inner class Errors {
+
+      @Test
+      fun `should put message on DLQ if NOMIS returns error`() = runTest {
+        nomisMockServer.stubGetPhysicalAttributes(NOT_FOUND)
+
+        sendPhysicalAttributesChangedEvent()
+
+        await untilAsserted {
+          assertThat(
+            awsSqsPrisonPersonOffenderEventDlqClient.countAllMessagesOnQueue(prisonPersonQueueOffenderEventsDlqUrl).get(),
+          ).isEqualTo(1)
+        }
+        nomisMockServer.verify(getRequestedFor(urlPathEqualTo("/prisoners/A1234AA/physical-attributes")))
+        dpsPrisonPersonServer.verify(0, putRequestedFor(urlPathEqualTo("/sync/prisoners/A1234AA/physical-attributes")))
+        verify(telemetryClient).trackEvent(
+          eq("physical-attributes-synchronisation-error"),
+          check {
+            assertThat(it["offenderNo"]).isEqualTo("A1234AA")
+            assertThat(it["bookingId"]).isEqualTo("12345")
+            assertThat(it["error"]).contains("Not Found")
+          },
+          isNull(),
+        )
+      }
+
+      @Test
+      fun `should put message on DLQ if DPS returns error`() = runTest {
+        nomisMockServer.stubGetPhysicalAttributes(aPrisonerPhysicalAttributesResponse())
+        dpsPrisonPersonServer.stubSyncPrisonPerson(BAD_REQUEST)
+
+        sendPhysicalAttributesChangedEvent()
+
+        await untilAsserted {
+          assertThat(
+            awsSqsPrisonPersonOffenderEventDlqClient.countAllMessagesOnQueue(prisonPersonQueueOffenderEventsDlqUrl).get(),
+          ).isEqualTo(1)
+        }
+        nomisMockServer.verify(getRequestedFor(urlPathEqualTo("/prisoners/A1234AA/physical-attributes")))
+        dpsPrisonPersonServer.verify(putRequestedFor(urlPathEqualTo("/sync/prisoners/A1234AA/physical-attributes")))
+        verify(telemetryClient).trackEvent(
+          eq("physical-attributes-synchronisation-error"),
+          check {
+            assertThat(it["offenderNo"]).isEqualTo("A1234AA")
+            assertThat(it["bookingId"]).isEqualTo("12345")
+            assertThat(it["error"]).contains("Bad Request")
+          },
+          isNull(),
+        )
+      }
+
+      @Test
+      fun `should put message on DLQ if booking has no physical attributes`() = runTest {
```

Comment:
why is this an error?

Reply:
We shouldn't be receiving events for bookings without any physical attributes - but I chose not to assume that in the code, so had to handle the error scenario. Should never happen.